File: backend/core/mail.py
import smtplib
from email.mime.text import MIMEText
import os

def send_code(receiver_email, verification_code, request_type):
    # 邮件服务器设置
    smtp_server = os.getenv('SMTP_SERVER')
    smtp_port = os.getenv('SMTP_PORT')
    sender_email = os.getenv('SMTP_SENDER')
    sender_password = os.getenv('SMTP_PASSWORD')

    # 根据 request_type 设置邮件内容
    if request_type == 'signup':
        email_body = f'Your registration verification code is {
            verification_code}'
    elif request_type == 'reset':
        email_body = f'Your password reset verification code is {
            verification_code}'
    else:
        email_body = f'Your verification code is {verification_code}'

    # 创建邮件内容
    message = MIMEText(email_body, 'plain', 'utf-8')
    message['From'] = sender_email
    message['To'] = receiver_email
    message['Subject'] = 'SeiunSodou Verification Code'

    # 连接邮件服务器并发送邮件
    server = smtplib.SMTP(smtp_server, smtp_port)
    server.starttls()
    server.login(sender_email, sender_password)
    server.sendmail(sender_email, [receiver_email], message.as_string())
    server.quit()
    return True

File: src/server/utils/mail.ts
import { createTransport } from "nodemailer";
import SMTPTransport from 'nodemailer/lib/smtp-transport'

const RuntimeConfig = useRuntimeConfig();

const options: SMTPTransport.Options = {
    host: RuntimeConfig.SMTP_SERVER,
    port: Number(RuntimeConfig.SMTP_PORT),
    secure: false, // 如果需要 SSL/TLS，请设置为 true
    auth: {
        user: RuntimeConfig.SMTP_SENDER,
        pass: RuntimeConfig.SMTP_PASSWORD
    }
}

/**
 * 发送验证码邮件的异步函数
 * 
 * @param {string} receiverEmail - 收件人的电子邮件地址
 * @param {number} verificationCode - 验证码
 * @param {string} requestType - 请求类型 ('signup' 或 'reset')
 * @returns {Promise<SMTPTransport.SentMessageInfo>} - 返回一个包含邮件发送信息的 Promise 对象，如果发送失败则返回错误信息
 */
export async function sendCode(
    receiverEmail: string,
    verificationCode: number,
    requestType: 'signup' | 'reset' | string
): Promise<SMTPTransport.SentMessageInfo> {

    // 根据 requestType 设置邮件内容
    let emailBody: string;
    if (requestType === 'signup') {
        emailBody = `Your registration verification code is ${verificationCode}`;
    } else if (requestType === 'reset') {
        emailBody = `Your password reset verification code is ${verificationCode}`;
    } else {
        emailBody = `Your verification code is ${verificationCode}`;
    }

    // 创建邮件传输器
    let transporter = createTransport(options);

    // 设置邮件选项
    let mailOptions = {
        from: `"星雲草堂SeiunSodou" <${RuntimeConfig.SMTP_SENDER}>`,
        to: receiverEmail,
        subject: 'SeiunSodou Verification Code',
        text: emailBody,
    };

    // 发送邮件
    try {
        let info = await transporter.sendMail(mailOptions);
        return info;
    } catch (error) {
        console.error(error);
        throw error;
    }
}